File: solutions/2960. Count Tested Devices After Test Operations/2960.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int countTestedDevices(const vector<int>& batteryPercentages) {
    int ans = 0;

    for (const int batteryPercentage : batteryPercentages)
      if (batteryPercentage -  ans > 0)
        ++ans;

    return ans;
  }
};

File: solutions/2960. Count Tested Devices After Test Operations/2960.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def countTestedDevices(self, batteryPercentages: List[int]) -> int:
    ans = 0

    for batteryPercentage in batteryPercentages:
      if batteryPercentage - ans > 0:
        ans += 1

    return ans